lib/win32/certstore.rb

```diff
@@ -118,8 +119,15 @@ def self.finalize(certstore_handler)
     attr_reader :certstore_handler
 
     # To open certstore and return open certificate store pointer
-    def open(store_name)
-      certstore_handler = CertOpenStore(CERT_STORE_PROV_SYSTEM, 0, nil, CERT_SYSTEM_STORE_LOCAL_MACHINE, wstring(store_name))
+
+    def open(store_name, store_location)
+      which_store_location = if store_location == "CurrentUser"
+                               CERT_SYSTEM_STORE_CURRENT_USER
+                             else
+                               CERT_SYSTEM_STORE_LOCAL_MACHINE
+                             end
```

Comment:
I'd sort of say do away with this remapping of your store_location to which_store_location and just use the windows constants directly.